src/arrayMethodSort.js

```diff
@@ -4,9 +4,32 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction = additionalCompareFunction) {
+    let isChanged = null;
+
+    do {
+      isChanged = false;
+
+      for (let i = 0; i < this.length - 1; i++) {
+        if (compareFunction(this[i], this[i + 1]) > 0) {
+          const firstElement = this[i];
+
+          this[i] = this[i + 1];
+          this[i + 1] = firstElement;
+
+          isChanged = true;
+        }
+      }
+    } while (isChanged);
+
+    return this;
   };
 }
 
+function additionalCompareFunction(a, b) {
```

Comment:
The comparison function(`additionalCompareFunction`) for the sort method must return one of three values: 1, 0, or -1. This is required for correct comparison of array elements

```
if (String(a) > String(b)) {
  return 1;
}
  
if (===) {
}
  
if (<) {
}
```